src/irmin-pack/s.ml

```diff
@@ -55,7 +55,40 @@ module type Specifics = sig
   (** [flush t] flush read-write pack on disk. Raises [RO_Not_Allowed] if called
       by a readonly instance.*)
 
-  val gc : ?unlink:bool -> repo -> commit_key -> unit
+  val gc :
+    ?unlink:bool ->
+    throttle:[ `Skip | `Block ] ->
+    repo ->
+    commit_key ->
+    unit Lwt.t
+
+  type generation = int
+  type pid = int
+
+  val start_gc :
+    throttle:[ `Block | `Skip ] ->
```

Comment:
I think "throttle" is the wrong word. Normally "throttle" would be something you do to reduce the speed of some process. Here, I think we are trying to say what should happen if GC is already running? So we will not "throttle" anything. In this case, a labelled argument ~if_gc_running:[`Block | `Skip] (or similar) would be clearer in my opinion.